appdev/guides/dependencies.rst

```diff
@@ -0,0 +1,136 @@
+
+Optionally, configure the compilation by adding ``build_args``, which may look like this:
+
+.. code-block:: JSON
+   :emphasize-lines: 7-12
+
+    {
+      "template": "cmake",
+      "dir": "build/app",
+      "libraries": {
+        "LIBNAME": {
+          "template": "cmake",
+          "build_args": [
+            "-DBUILD_EXAMPLES=OFF",
+            "-DBUILD_DOCS=OFF",
+            "-DBUILD_TESTS=OFF",
+            "-DBUILD_SHARED_LIBS=OFF"
+          ]
+        }
+      }
+    }
+
+Build arguments are usually project specific. Therefore, study the library's build instructions and also look for ``option`` settings in its ``CMakeLists.txt`` (if CMake is used). The default build directory is ``build/LIBNAME/ARCHITECTURE``. Therefore builds for different architectures can exist in parallel.
+
+To actually build the library run ``clickable build-libs``. You may want to build it for the desktop, too, via ``clickable build-libs --arch amd64``. Don't forget to mention this step in your README!
+
+See how `Teleports clickable.json <https://gitlab.com/ubports/apps/teleports/blob/master/clickable.json#L14>`_ uses the libraries feature to build its dependency tdlib.
+
+Pre-built
+^^^^^^^^^
+Note, that a pre-built library is usually only available as a shared object, that needs to be linked dynamically (no static linking). Furthermore, it may contain components that you don't need, resulting in a bloated app. It may even miss something, that you could achieve by compiling it yourself. Sometimes, a library is available in the Ubuntu Repositories, but is not installable for the architecture you need (likely ``armhf``). In this case you have to compile the library as described above.
+
+If the library is available in the Ubuntu Repositories, you can add it to the dependencies list, like this:
+
+.. code-block:: JSON
+   :emphasize-lines: 3-5
+
+    {
+      "template": "cmake",
+      "dependencies_target": [
+        "libsomething-dev"
+      ]
+    }
+
+Clickable will install the specified package automatically for the target architecture inside the build container. An example can be found in `Guitar Tools' clickable.json <https://github.com/t-mon/guitar-tools/blob/master/clickable.json#L4>`_.
+
+If the library is not in the Ubuntu Repositories, but in a PPA, you can add the PPA to the clickable.json, too. For example:
+
+.. code-block:: JSON
+   :emphasize-lines: 3-5
+
+    {
+      "template": "cmake",
+      "dependencies_ppa": [
+        "ppa:someone/libsomething"
+      ],
+      "dependencies_target": [
+        "libsomething-dev"
+      ]
+    }
+
+Otherwise you may need to add a script to download the pre-built library.
+
+Using
+-----
+You may need to specify the include directory, where the compiler can find the headers, and you need to link the library itself against your app's binary (if it is not a pure header library).
```

Comment:
I'm not entirely sure what a pure header library is. If that detail is required, I'd suggest expanding on it. If not, it can be removed. Maybe flip what is inside and outside of the parentheses?

The first comma here is not needed.

Reply:
A pure header library does only container header files. The library has no source files that are meant to be compiled into a library binary. You can use the whole library by including the headers. For example some Boost libraries are like that.

Do you think that is clearer now in the text?